dart/lib/src/http_client/failed_request_client.dart

```diff
@@ -0,0 +1,243 @@
+  ///   because the connection was interrupted.
+  final bool captureFailedRequests;
+
+  /// Configures up to which size request bodies should be included in events.
+  /// This does not change wether an event is captured.
+  final MaxRequestBodySize maxRequestBodySize;
+
+  /// Describes which HTTP status codes should be considered as a failed
+  /// requests.
+  ///
+  /// Per default no status code is considered a failed request.
+  final List<SentryStatusCode> failedRequestStatusCodes;
+
+  final bool sendDefaultPii;
+
+  @override
+  Future<StreamedResponse> send(BaseRequest request) async {
+    int? statusCode;
+    Object? exception;
+    StackTrace? stackTrace;
+
+    final stopwatch = Stopwatch();
+    stopwatch.start();
+
+    try {
+      final response = await _client.send(request);
+      statusCode = response.statusCode;
+      return response;
+    } catch (e, st) {
+      exception = e;
+      stackTrace = st;
+      rethrow;
+    } finally {
+      stopwatch.stop();
+
+      // If captureFailedRequests is true, there statusCode is null.
+      // So just one of these blocks can be called.
+
+      if (captureFailedRequests && exception != null) {
+        await _captureEvent(
+          exception: exception,
+          stackTrace: stackTrace,
+          request: request,
+          requestDuration: stopwatch.elapsed,
+        );
+      } else if (failedRequestStatusCodes.containsStatusCode(statusCode)) {
+        // Capture an exception if the status code is considered bad
+        await _captureEvent(
+          request: request,
+          reason: failedRequestStatusCodes.toDescription(),
+          requestDuration: stopwatch.elapsed,
+        );
+      }
+    }
+  }
+
+  @override
+  void close() {
+    // See https://github.com/getsentry/sentry-dart/pull/226#discussion_r536984785
+    _client.close();
+  }
+
+  // See https://develop.sentry.dev/sdk/event-payloads/request/
+  Future<void> _captureEvent({
+    Object? exception,
+    StackTrace? stackTrace,
+    String? reason,
+    required Duration requestDuration,
+    required BaseRequest request,
+  }) {
+    // As far as I can tell there's no way to get the uri without the query part
+    // so we replace it with an empty string.
+    final urlWithoutQuery = request.url.replace(query: '').toString();
+
+    final query = request.url.query.isEmpty ? null : request.url.query;
+
+    final sentryRequest = SentryRequest(
+      method: request.method,
+      headers: sendDefaultPii ? request.headers : null,
+      url: urlWithoutQuery,
+      queryString: query,
+      cookies: sendDefaultPii ? request.headers['Cookie'] : null,
+      data: _getDataFromRequest(request),
+      other: {
+        'content_length': request.contentLength.toString(),
+        'duration': requestDuration.toString(),
+      },
+    );
+
+    final mechanism = Mechanism(
+      type: 'SentryHttpClient',
+      handled: true,
```

Comment:
setting to `true` or not setting at all is the same, we could remove and save a few chars